### PR TITLE
Remove controllers not used by CAPI. Update PodCIDR controller to use CAPI resources as source of truth.

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/16000
Towards https://github.com/giantswarm/giantswarm/issues/16005

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by kubasobon** [comment]
It still creates App CRs, but requires namespace to be created beforehand.

**Comment by JosephSalisbury** [comment]
yo @kubasobon mr @TheoBrigitte wants you to give a more descriptive title for this PR

**Comment by kubasobon** [comment]
Tested. This is enough to handle CAPI clusters and install apps on them.

**Comment by rossf7** [comment]
@kubasobon Just a heads up that I plan to work on it this week.

For https://github.com/giantswarm/roadmap/issues/320 I'll be creating a new cluster-apps-operator based on your changes here.

File: helm/cluster-operator/templates/configmap.yaml [hunk 1]
```diff
@@ -15,16 +15,14 @@ data:
...
     guest:
       cluster:
+        baseDomain: '{{ .Values.baseDomain }}'
         calico:
           subnet: '{{ .Values.cni.subnet }}'
           cidr: '{{ .Values.cni.mask }}'
         kubernetes:
           api:
-            clusterIPRange: '{{ .Values.kubernetes.api.clusterAPIRange }}'
+            clusterIPRange: '{{ .Values.kubernetes.api.clusterIPRange }}'
```
Comment (anvddriesch) [inline helm/cluster-operator/templates/configmap.yaml:24]: Did you test this and did it work before?
`cluster-operator-cluster-operator-3.7.2-capi` is crashlooping now on `ginger` and I see this error:
```
panic: {"kind":"invalidConfigError","annotation":"invalid CIDR address: ","stack":[{"file":"/root/project/service/controller/key/key.go","line":32},{"file":"/root/project/service/service.go","line":122}]}
```

The reason seems to be that the value for `clusterIPRange` in the configmap is empty.

Comment (kubasobon) [reply helm/cluster-operator/templates/configmap.yaml:24]: I tested it before I fixed it, but I missed a spot, namely `helm/values.yaml`. Fixed by https://github.com/giantswarm/cluster-operator/pull/1357/commits/9f13fb7841d03486fc5cc87334fe650e81d0febe.
I should probably re-release. What would the version be  `3.7.2-capi1`? `3.7.3-capi`?

Comment (anvddriesch) [reply helm/cluster-operator/templates/configmap.yaml:24]: Awesome :) 
I don't have any preference there. Just go with what sounds right to you.

Comment (kubasobon) [reply helm/cluster-operator/templates/configmap.yaml:24]: [v3.7.2-capi-1](https://github.com/giantswarm/cluster-operator/releases/tag/v3.7.2-capi-1) is now a thing.

Comment (anvddriesch) [reply helm/cluster-operator/templates/configmap.yaml:24]: Nice, I'll replace it in the release.

Comment (kubasobon) [reply helm/cluster-operator/templates/configmap.yaml:24]: Eh, seems our CI does not like what I did one bit: https://app.circleci.com/pipelines/github/giantswarm/cluster-operator/2517/workflows/b0eed0ef-8966-46ab-9921-771a1f101105/jobs/12571
I'll have to create another release and remember about `pkg/project/project.go` this time :sob:

**Comment by piontec** [comment]
This seems to be a bit stuck since May. What's the current status? @rossf7 is there anything left to do on managed apps side?

**Comment by MarcelMue** [comment]
I believe we should close this PR. We have https://github.com/giantswarm/cluster-apps-operator for this purpose now.

**Comment by kubasobon** [comment]
Closing since cluster-operator is no longer used in the [CAPA release](https://github.com/giantswarm/releases/blob/master/aws/v20.0.0/release.yaml).

**Comment by rossf7** [comment]
@kubasobon Thanks for closing.

@piontec Final steps are here. https://github.com/giantswarm/roadmap/issues/320#issuecomment-905457199

**Pull request closed without merging**